main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -360,7 +360,7 @@ object Ast {
   /**
     * Contains the edges held in a single constraint.
     */
-  case class MultiEdge(head: Name.Pred, positives: Set[(Name.Pred, SourceLocation)], negatives: Set[(Name.Pred, SourceLocation)])
+  case class MultiEdge(head: (Name.Pred, Int), positives: Set[(Name.Pred, Int, SourceLocation)], negatives: Set[(Name.Pred, Int, SourceLocation)])
```

Comment:
Do you want to carry the type here instead and switch to a non-set representation?

Reply:
I am a 100% certain I did this already but I guess a made git forget about it somehow :l

Reply:
Git strikes again!